Src/AutoFixture.NUnit3/MutableNameTestMethodBuilder.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using System.Diagnostics.CodeAnalysis;
+using NUnit.Framework.Internal.Builders;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    /// <summary>
+    /// Utility used to create a <see cref="TestMethod"/> instance.
+    /// </summary>
+    /// <seealso cref="ITestMethodBuilder" />
+    public class MutableNameTestMethodBuilder : ITestMethodBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MutableNameTestMethodBuilder"/> class.
+        /// </summary>
+        public MutableNameTestMethodBuilder() { }
+
+        /// <inheritdoc />
+        public virtual TestMethod Build(IMethodInfo method, Test suite, Func<object[]> argFactory, int autoDataStartIndex)
+        {
+            if (method == null)
+            {
+                throw new ArgumentNullException(nameof(method));
+            }
+
+            if (argFactory == null)
+            {
+                throw new ArgumentNullException(nameof(argFactory));
+            }
+
+            return new NUnitTestCaseBuilder().BuildTestMethod(method, suite, GetParametersForMethod(method, argFactory, autoDataStartIndex));
+        }
+
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "This method is always expected to return an instance of the TestCaseParameters class.")]
+        private TestCaseParameters GetParametersForMethod(IMethodInfo method, Func<object[]> argFactory, int autoDataStartIndex)
+        {
+            try
+            {
+                var parameterValues = argFactory();
+                return GetParametersForMethod(method, parameterValues.ToArray(), autoDataStartIndex);
+            }
+            catch (Exception ex)
```

Comment:
Please add a test for exception case. Ensure to add tests for both the strategies.